sycl/doc/extensions/experimental/sycl_ext_oneapi_memory_export.asciidoc

```diff
@@ -0,0 +1,388 @@
+resource handles that can be exported by this extension. The `opaque_fd` and
+`win32_nt_handle` values are used during allocation of exportable memory to
+indicate the type of handle that will later be returned by the
+`export_device_mem_handle` function.
+
+The `opaque_fd` handle type corresponds to a POSIX file descriptor, which is
+represented by an `int`.
+
+The `win32_nt_handle` handle type corresponds to a Windows NT handle, which is
+represented by a `void *`.
+
+The `exported_mem` template struct is specialized for each of the
+`external_mem_handle_type` values. Each specialization defines a `type` alias
+that represents the type of the exported memory handle for that specific handle.
+
+The `exported_mem_t` type alias allows the user to easily obtain the type of
+the exported memory handle based on the `external_mem_handle_type` template
+parameter.
+
+=== API of the extension
+
+```c++
+
+namespace sycl::ext::oneapi::experimental {
+
+void *alloc_exportable_device_mem(
+    size_t alignment, size_t size,
+    external_mem_handle_type externalMemHandleType,
+    const sycl::device &syclDevice, const sycl::context &syclContext,
+    const property_list& propList = {});
+
+void *alloc_exportable_device_mem(
+    size_t alignment, size_t size,
+    external_mem_handle_type externalMemHandleType,
+    const sycl::queue &syclQueue,
+    const property_list& propList = {});
+}
+```
+
+The `alloc_exportable_device_mem` function allocates memory on the device marked
+as having the ability to later export that memory to an external memory resource
+handle.
+
+Memory allocated through this function must only be freed using
+`free_exportable_mem`. Using `sycl::free` to deallocate memory allocated with
+this function results in undefined behavior.
+
+With the exception of the `sycl::free` function from the core SYCL
+specification, pointers to memory allocated through this function may be passed
+to any core SYCL specification API accepting device USM memory pointers.
+
+Memory allocated through this function is only available on device.
+
+Memory allocated through this function has a linear memory layout on the device 
+(which is the same as memory allocated by other USM allocation functions like 
+`sycl::malloc_device`).
+
+Zero or more properties can be passed in the `propList` parameter via an
+instance of `sycl::property_list`. Currently, this extension does not define
+any properties that can be used with this function, so the `propList` parameter
+is ignored and reserved for future use.
+
+The passed `externalMemHandleType` must be supported by the operating system
+running the SYCL application. Passing an `externalMemHandleType` that is not
+supported by the operating system running the SYCL application results in
+undefined behavior. The `win32_nt_handle` handle type is only supported on
+Windows operating systems, while the `opaque_fd` handle type is only supported
+on POSIX compliant operating systems.
+
```

Comment:
This paragraph can be deleted now because it's already covered in the next one.  Also, note that the behavior when passing a memory handle type that is unsupported by the O/S is no longer "undefined".  The behavior is now documented to throw an exception.

Reply:
Removed.